core/shared/src/main/scala-2.11-2.12/ChunkBuilder.scala

```diff
@@ -79,28 +81,52 @@ private[zio] object ChunkBuilder {
    * values.
    */
   final class Boolean extends ChunkBuilder[SBoolean] { self =>
-    private val arrayBuilder: ArrayBuilder[SBoolean] = {
-      new ArrayBuilder.ofBoolean
+
+    private val bytes: ArrayBuffer[SByte] = ArrayBuffer()
```

Comment:
It looks like `ArrayBuffer` is not specialized so this will box primitive values. Can we instead use an `ArrayBuilder.ofByte` or even a `ChunkBuilder.Byte`? I think this would require maintaining a `var` with the last "unfinished" byte that we manipulate until we are done and add to the `ChunkBuilder` either when it is done or when `result` is called.

Reply:
> 
> 
> It would also be good to add some tests here, for example that when we use the new `ChunkBuilder` to build a chunk from a list of random booleans and turn it back into a list that we get back the list unchanged.

It looks like we already have such test:
https://github.com/whysoserious/zio/blob/master/core-tests/shared/src/test/scala/zio/ChunkBuilderSpec.scala#L17-L23

Btw, that's great that we are using property based tests in the project. They already helped me to catch a few subtle bugs in my impl.